.circleci/config.yml

```diff
@@ -574,76 +627,148 @@ jobs:
       - slack/status:
           webhook: << parameters.slack-webhook >>
 
+  desktop-build-source:
```

Comment:
Building the desktop client + server source can take as long ~10 minutes ([link](https://app.circleci.com/pipelines/github/Automattic/wp-calypso/65123/workflows/69f62ec9-7575-4f8c-a4f2-1d734f48d4f5/jobs/787903)). 

As with the wp-desktop build today, we try to optimize this by building the source once, which is then sent to a downstream job for each platform to compile the final application bundle/binary.